cpp/ql/src/Critical/MissingCheckScanf.ql

```diff
@@ -112,10 +112,21 @@ BasicBlock blockGuardedBy(int value, string op, ScanfFunctionCall call) {
   )
 }
 
+predicate isDeallocationAccess(Access access) {
+  exists(FunctionCall deallocCall, DeallocationFunction deallocFunc |
+    deallocCall.getTarget() = deallocFunc
+  |
+    deallocCall.getArgument(deallocFunc.getFreedArg()) = access
+  )
```

Comment:
Do we need this first case?  A call to a `DeallocationFunction` should be a `DeallocationExpr` and so covered by the second case.

Reply:
That wasn't clear to me. I give it a go.

Reply:
Done. Thanks for this.